File: app/concepts/matestack/ui/core/form/submit/submit.rb
require_relative './base'

module Matestack::Ui::Core::Form::Submit
  class Submit < Base

    vue_js_component_name "matestack-ui-core-form-submit"

    def response
      span attributes: submit_attributes do
        yield_components
      end
    end

  end
end

File: app/concepts/matestack/ui/core/form/submit/submit.js
import Vue from "vue/dist/vue.esm";

import componentMixin from "../../component/component";

const componentDef = {
  mixins: [componentMixin],
  data() {
    return {};
  },
  methods: {
    loading: function(){
      return this.$parent.loading;
    }
  }
}

let component = Vue.component("matestack-ui-core-form-submit", componentDef);

export default componentDef;
